### PR TITLE
Fixed some possible null pointer crashes and race condition issues

### PR DESCRIPTION
**Pull request by bert-and-ernie**
* The MediaStream in LocalStream can be disposed before publish, ConferenceClient checks if not already disposed before publish.
* Stream instance in ConferencePeerConnectionChannel can be null in function ConferenceClient.subscribe()
* Added extra checks and synchronized block in ConferenceClient.onStreamUpdated() when using remoteStreams

### CODE REVIEW

**Review by jianjunz — APPROVED** [review]
Thanks for your change.

File: src/sdk/base/src/main/java/owt/base/LocalStream.java [hunk 1]
```diff
@@ -77,4 +77,8 @@ public void dispose() {
         mediaStream.dispose();
         mediaStream = null;
     }
+
+    public boolean disposed() {
```
Comment (jianjunz) [inline src/sdk/base/src/main/java/owt/base/LocalStream.java:81]: It would be better to have descriptions for public functions.

Comment (bert-and-ernie) [reply src/sdk/base/src/main/java/owt/base/LocalStream.java:85]: Hello jianjunz, I've added a description for the function.

Comment (jianjunz) [reply src/sdk/base/src/main/java/owt/base/LocalStream.java:85]: Thanks.

**Review by jianjunz — DISMISSED** [review]

**Pull request closed without merging**